### PR TITLE
Change `getPDFFileNameFromURL` to ignore `data:` URLs for performance reasons (issue 8263)

### LINKED ISSUE #8263: Don't guess file name from non-http/file URLs
From e-mail:

> pdf.js tries to find a nice filename for the document. this takes an incredible amount of time when a data-uri document is displayed

data: or blob: schemas doesn't any useful info to guess title, so viewer.js:1709-1710 / getPDFFileNameFromURL may skip regex if non-http/file URL is used.

### PR DESCRIPTION
**Pull request by Snuffleupagus**
*Please refer to the individual commit messages.*

Tentatively fixes #8263.

### CODE REVIEW

File: web/ui_utils.js [hunk 1]
```diff
@@ -355,22 +355,25 @@ function noContextMenuHandler(e) {
...
+ * @param {string} url - The original PDF location.
+ * @param {string} defaultFilename - The value returned if the filename is
+ *   unknown, or the protocol is unsupported.
+ * @returns {string} Guessed PDF filename.
  */
-function getPDFFileNameFromURL(url, defaultFilename) {
-  if (typeof defaultFilename === 'undefined') {
-    defaultFilename = 'document.pdf';
+function getPDFFileNameFromURL(url, defaultFilename = 'document.pdf') {
+  if (url.indexOf('data:') === 0) {
```
Comment (yurydelendik) [inline web/ui_utils.js]: Per https://tools.ietf.org/html/rfc3986#section-3.1 , schemes are case-insensitive. I recommend to create the isDataSchema(url) function above and also skip allowed whitespaces, something like:
```
var i = 0; while (i < url.length && url[i].trim() == '') i++;
return url.substr(i, 5).toLowerCase() === 'data:';
```

**Review by yurydelendik — APPROVED** [review]
Looks good with the change below

File: web/ui_utils.js (hunk 1 above)
Comment (Snuffleupagus) [inline web/ui_utils.js]: Thank you for the review!
I've made the requested changes, and also added a few more unit-tests for the (whitespace) case.
